sdk-design-doc-template.md

```diff
@@ -0,0 +1,83 @@
+# SDK Design Doc Template
+
+Template for SDK design docs that will be checked into a SDK design docs repo.
```

Comment:
We need a better definition here. Should such document be created whenever something is added to protobuf?

Reply:
How about:

This is the template to use for Hiero SDK design documents. These documents should be generated for every new feature looking to be added to the Hiero SDKs. These features can include, but are not limited to, new HIP implementations, internal SDK processing changes, other new or changed APIs, etc. Any change that should be applied to all Hiero SDKs should use this design doc template to describe the proposed changes and test plan.